app/controllers/application_controller.rb

```diff
@@ -83,6 +83,11 @@ def doorkeeper_unauthorized_render_options(error: nil)
     session[:locale] = nil
     set_locale
 
+    if current_user.should_claim_wca_id?
+      flash[:info] = t('devise.after_sign_in.wca_claim_alert', claim_wca_id_path: profile_claim_wca_id_url).html_safe
+      current_user.update(should_claim_wca_id: false)
```

Comment:
This `update` call returns a boolean which is currently ignored. You have to put this into an `if` somehow, to reasonably react to the case where the update operation in the DB failed.

Reply:
Easiest case: Change the outer `if` to `if current_user.should_claim_wca_id? && current_user.update(...)`.
But this is potentially dangerous seeing that it does not allow you to distinguish between "`else` because `current_user.should_claim_wca_id?` is false" and "`else` because `should_claim_wca_id?` is true but the subsequent `update` failed".